src/components/lib/cos_kernel_api.c

```diff
@@ -421,7 +422,7 @@ cos_captbl_alloc(struct cos_compinfo *ci)
 	assert(ci);
 
 	if (__alloc_mem_cap(ci, CAP_CAPTBL, &kmem, &cap)) return 0;
-	if (call_cap_op(ci->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, cap, ci->pgtbl_cap, kmem, 0)) BUG();
+	if (call_cap_op(ci->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, cap, __compinfo_metacap(ci)->local_mi.pgtbl_cap, kmem, 0)) BUG();
```

Comment:
More concrete, `SELF_PT` is a fixed value while `ci->pgtbl_cap` is a variable.
But you are right. both `ci->pgtbl_cap` in manager's cap table and `SELF_PT` in ci's cap table refers to the same page table